page.codeberg.SOrg.DogGTK.json

```diff
@@ -0,0 +1,44 @@
+{
+    "app-id" : "page.codeberg.SOrg.DogGTK",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "45",
+    "sdk" : "org.gnome.Sdk",
+    "sdk-extensions" : [
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command" : "doggtk",
+    "finish-args" : [
+        "--socket=wayland",
+        "--socket=fallback-x11",
+        "--share=ipc",
+        "--device=dri"
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin"
+    },
+    "cleanup" : [
+        "/include",
+        "/lib/pkgconfig",
+        "/man",
+        "/share/doc",
+        "/share/gtk-doc",
+        "/share/man",
+        "/share/pkgconfig",
+        "*.la",
+        "*.a"
+    ],
+    "modules" : [
+        {
+            "name" : "doggtk",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://codeberg.org/SOrg/DogGTK.git",
```

Comment:
You're depending on external crates and got doesn't have vendoring. I doubt you'll be able to build this because cargo can't fetch them as flathub runs without network access.

You probably need to use https://github.com/flatpak/flatpak-builder-tools/tree/master/cargo to create a dependency manifest first

Reply:
ok, i'll try work on this now, sorry for how long it took to respond